htdocs/adherents/class/adherentstats.class.php

```diff
@@ -181,4 +182,56 @@ public function getAllByYear()
 
 		return $this->_getAllByYear($sql);
 	}
+
+
+	/**
+	 *	Return count of member by status group by adh type, total and average
+	 *
+	 * 	@return		array                   Array with
+	 */
+	public function countMembersByTypeAndStatus()
+	{
+		global $user;
+
+		$now = dol_now();
+
+		$sql = "SELECT COALESCE(d.fk_adherent_type, 'total') as fk_adherent_type, t.libelle as label";
+		$sql .= ", COUNT(CASE WHEN d.statut = ".Adherent::STATUS_DRAFT." THEN 'members_draft' ELSE NULL END) as members_draft";
+		$sql .= ", COUNT(CASE WHEN d.statut = ".Adherent::STATUS_VALIDATED."  AND (d.datefin IS NULL AND t.subscription = '1') THEN 'members_pending' ELSE NULL END) as members_pending";
+		$sql .= ", COUNT(CASE WHEN d.statut = ".Adherent::STATUS_VALIDATED."  AND (d.datefin >= '".$this->db->idate($now)."' OR t.subscription = 0) THEN 'members_uptodate' ELSE NULL END) as members_uptodate";
+		$sql .= ", COUNT(CASE WHEN d.statut = ".Adherent::STATUS_VALIDATED."  AND (d.datefin < '".$this->db->idate($now)."' AND t.subscription = 1) THEN 'members_expired' ELSE NULL END) as members_expired";
+		$sql .= ", COUNT(CASE WHEN d.statut = ".Adherent::STATUS_EXCLUDED." THEN 'members_excluded' ELSE NULL END) as members_excluded";
+		$sql .= ", COUNT(CASE WHEN d.statut = ".Adherent::STATUS_RESILIATED."  THEN 'members_resiliated' ELSE NULL END) as members_resiliated";
+		$sql .= " FROM ".MAIN_DB_PREFIX."adherent as d , ".MAIN_DB_PREFIX."adherent_type as t";
```

Comment:
If a type has no member at all, it seems this request will return nothing when need need a count per type, so if there is no member at all for this type, we need a result into the sql request. May be we need to convert the join into left join ?

Reply:
Done